### PR TITLE
Consistent naming for context gettable param queries.

### PR DESCRIPTION
**Pull request by paulidale**
All instances of `EVP_*_CTX_gettable_params` functions have been renamed
to `EVP_*_gettable_ctx_params`.  Except for the `EVP_MD` ones which were changed
already.

These functions do not take `EVP_*_CTX` arguments so their prior naming was
misleading.

- [x] documentation is added or updated
- [x] tests are added or updated

Conversion was mechanical:
```
sed -i -E 's/EVP_(KDF|MAC|CIPHER)_CTX_([a-z]{4,})_params/EVP_\1_\2_ctx_params/g'  `egrep -rlw 'EVP_(KDF|MAC|CIPHER)_CTX_[gs]ettable_params' .`
```

### CODE REVIEW

File: util/libcrypto.num [hunk 1]
```diff
@@ -4748,8 +4748,8 @@ EVP_PKEY_verify_init_ex                 4864	3_0_0	EXIST::FUNCTION:
 EVP_PKEY_verify_recover_init_ex         4865	3_0_0	EXIST::FUNCTION:
 EVP_PKEY_CTX_get_signature_md           4866	3_0_0	EXIST::FUNCTION:
 EVP_PKEY_CTX_get_params                 4867	3_0_0	EXIST::FUNCTION:
-EVP_PKEY_CTX_gettable_params            4868	3_0_0	EXIST::FUNCTION:
-EVP_PKEY_CTX_settable_params            4869	3_0_0	EXIST::FUNCTION:
+EVP_PKEY_gettable_ctx_params            4868	3_0_0	EXIST::FUNCTION:
+EVP_PKEY_settable_ctx_params            4869	3_0_0	EXIST::FUNCTION:
```
Comment (levitte) [inline util/libcrypto.num]: Side note: this one needs more investigation...

Comment (paulidale) [reply util/libcrypto.num]: It shouldn't have been renamed.  It takes a PKEY_CTX argument.
Fix pushed.

**Comment by levitte** [comment]
It's interesting to see how radically different the EVP_PKEY interface has become... It's not exactly discovery friendly. I will try to have a look at that at some point...

**Comment by paulidale** [comment]
Merged to master, thanks.

**Pull request closed without merging**